akka-sample-cqrs-scala/src/main/scala/sample/cqrs/Main.scala

```diff
@@ -49,56 +60,100 @@ object Main {
 
     // TODO use real replication strategy in real application
-    val keyspaceStmt = """
+    val keyspaceStmt =
+      """
       CREATE KEYSPACE IF NOT EXISTS akka_cqrs_sample
       WITH REPLICATION = { 'class' : 'SimpleStrategy', 'replication_factor' : 1 }
       """
 
     val offsetTableStmt =
       """
-      CREATE TABLE IF NOT EXISTS akka_cqrs_sample.offsetStore (
-        eventProcessorId text,
-        tag text,
-        timeUuidOffset timeuuid,
-        PRIMARY KEY (eventProcessorId, tag)
+      CREATE TABLE IF NOT EXISTS akka_cqrs_sample.offset_store (
+        projection_name text,
+        partition int,
+        projection_key text,
+        offset text,
+        manifest text,
+        last_updated timestamp,
+        PRIMARY KEY ((projection_name, partition), projection_key)
       )
       """
 
     // ok to block here, main thread
     Await.ready(session.executeDDL(keyspaceStmt), 30.seconds)
+    system.log.info("Created akka_cqrs_sample keyspace")
     Await.ready(session.executeDDL(offsetTableStmt), 30.seconds)
+    system.log.info("Created akka_cqrs_sample.offset_store table")
+
   }
 
 }
 
 object Guardian {
+
   def apply(): Behavior[Nothing] = {
     Behaviors.setup[Nothing] { context =>
       val system = context.system
+
       val settings = EventProcessorSettings(system)
+
       val httpPort = context.system.settings.config.getInt("shopping.http.port")
 
       ShoppingCart.init(system, settings)
 
       if (Cluster(system).selfMember.hasRole("read-model")) {
-        // FIXME, the tables may not be created yet, send a start message they're done
-        EventProcessor.init(
-          system,
-          settings,
-          tag => new ShoppingCartEventProcessorStream(system, system.executionContext, settings.id, tag))
+        def createProjectionFor(tag: String) = {
+          val sourceProvider = EventSourcedProvider
+            .eventsByTag[ShoppingCart.Event](
+              system = system,
+              readJournalPluginId = CassandraReadJournal.Identifier,
+              tag = tag
+            )
+          CassandraProjection
+            .atLeastOnce(
+              projectionId = ProjectionId("shopping-carts", tag),
+              sourceProvider,
+              handler = new ShoppingCartProjectionHandler(tag, system)
+            )
+            .withSaveOffset(100, 500.millis)
+        }
+
+        // we only want to run the daemon processes on the read-model nodes
+        val shardingSettings = ClusterShardingSettings(system)
+        val shardedDaemonProcessSettings =
+          ShardedDaemonProcessSettings(system)
+            .withShardingSettings(shardingSettings.withRole("read-model"))
+
+        ShardedDaemonProcess(system).init(
+          name = "ShoppingCartProjection",
+          settings.parallelism,
+          // factory for the behaviors
+          { n =>
+            val tag = s"${settings.tagPrefix}-$n"
+            ProjectionBehavior(createProjectionFor(tag))
```

Comment:
shall we move the `tag` construction into `createProjectionFor` and make this a one-liner `n => createProjection(n)`?

Reply:
I was considering that but couldn't decide. Yes, let's do that.